lib/net.js

```diff
@@ -524,16 +524,8 @@ Socket.prototype._read = function(n) {
 
 Socket.prototype.end = function(data, encoding) {
   stream.Duplex.prototype.end.call(this, data, encoding);
```

Comment:
While we are at it, can you add a callback to `end()`? that's the full signature for streams (https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback).

Reply:
yup, done!